File: stack/cpp/b2812.cpp
#include <iostream>
#include <string>
#include <vector>
int N, K;
std::string numbers;
std::vector<char> result;

int main() {
	std::cin >> N >> K;
	std::cin >> numbers;

	int cnt = N - K;
	result.reserve(cnt);
	result.push_back(numbers[0]);
	numbers.erase(numbers.begin());
	for (auto& number : numbers) {
		while (K and result.size() and number > result[result.size() - 1]) {
			result.pop_back();
			K--;
		}
		if (result.size() < cnt) {
			result.push_back(number);
			continue;
		}
		break;
	}
	for (int i = 0; i < result.size(); ++i) {
		std::cout << result[i];
	}
}

File: stack/python/b2812.py
N, K = map(int, input().split())
numbers = input()
cnt = N - K

check_stack = [numbers[0]]

for number in numbers[1:]:
    while K and check_stack and number > check_stack[-1]:
        K -= 1
        check_stack.pop()

    if cnt > len(check_stack):
        check_stack.append(number)
    else:
        break

print("".join(check_stack))